### ISSUE TITLE
[BUG]: group_features bug

### ISSUE THREAD

**Issue by moezali1** [pycaret/pycaret#3374]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

I notice the behavior of `group_features` in PyCaret 3.0 has changed. It now drops the original features which is incorrect. It shouldn't drop original features but instead just add newly created features to dataframe.

```
# load dataset
from pycaret.datasets import get_data
credit = get_data('credit')

# init setup
from pycaret.classification import *
clf1 = setup(data = credit, target = 'default', 
             group_features = ['BILL_AMT1', 'BILL_AMT2', 'BILL_AMT3', 'BILL_AMT4', 'BILL_AMT5', 'BILL_AMT6'])
```

![image](https://user-images.githubusercontent.com/54699234/223572700-15a17158-6ab9-410f-8256-ce50cb0a7a33.png)

Notice that all BILL_AMT... original features are gone.

### Reproducible Example

```python
see above
```

### Expected Behavior

see above

### Actual Results

```python-traceback
see above
```

### Installed Versions

rc9

**Comment by tvdboom** [comment]
I don't agree. The whole point of extracting the statistics from a group of features is to be able to remove a potentially large number f features without losing too much of its information. Why else would you create the new features if you keep the old?

**Comment by moezali1** [comment]
The original implementation was around the idea of new feature generation hence we were not removing the original one. What do we do in `polynomial_features`. Do we replace original columns or add new ^2 columns?

**Comment by tvdboom** [comment]
in polynomial we add new features. That case is different since you are looking for new non-linear correlations between the features. That's not the case here.

**Comment by mfahadakbar** [comment]
In my openion, when we make new features, we are assuming that new features will capture some thing extra that original features were not capturing (or were  not enough). So old features should be kept as a default behaviour. E.g. I have time lag features, apart from time lag I also want to group them with min , max, mean etc..it is very obvious that we should keep both as two type of feature are trying to capture different things...

There could also be practical reasons for that, e.g explainability around the original features .

However, It is not something that is always required. So I think the solution would be to keep the original features by default , and have an option to drop them off required.

**Comment by Yard1** [comment]
I think it would be reasonable to drop them by default, but give an option to keep them.

**Comment by tvdboom** [comment]
alright, I'll add the option, leaving the features by default

**Comment by moezali1** [comment]
tested using `master` branch on 3/15/2023. The issue has been fixed.
